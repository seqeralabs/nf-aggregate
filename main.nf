#!/usr/bin/env nextflow
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    seqeralabs/nf-aggregate
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Github : https://github.com/seqeralabs/nf-aggregate
----------------------------------------------------------------------------------------
*/

nextflow.enable.dsl = 2

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    CONFIG FILES
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

ch_multiqc_custom_config = params.multiqc_config ? Channel.fromPath(params.multiqc_config, checkIfExists: true) : Channel.empty()
ch_multiqc_logo          = params.multiqc_logo   ? Channel.fromPath(params.multiqc_logo, checkIfExists: true) : Channel.fromPath("$projectDir/assets/seqera_logo_colour.png", checkIfExists: true )

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    IMPORT FUNCTIONS / MODULES / SUBWORKFLOWS / WORKFLOWS
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

include { NF_AGGREGATE            } from './workflows/nf_aggregate'
include { PIPELINE_INITIALISATION } from './subworkflows/local/utils_nf_aggregate'

//
// WORKFLOW: Execute a single named workflow for the pipeline
//
workflow {

    //
    // SUBWORKFLOW: Run initialisation tasks
    //
    PIPELINE_INITIALISATION ()

    //
    // WORKFLOW: Run primary workflows for the pipeline
    //
    NF_AGGREGATE (
        PIPELINE_INITIALISATION.out.ids,
        ch_multiqc_custom_config,
        ch_multiqc_logo,
        params.seqera_api_endpoint,
        params.skip_run_gantt,
        params.skip_multiqc
    )
}

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    THE END
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/
