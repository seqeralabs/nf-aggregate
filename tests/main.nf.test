nextflow_pipeline {

    name "Test Pipeline main.nf"
    script "../main.nf"

    tag "pipeline"
    tag "pipeline_nf-aggregate"

    test("Should run without failures") {

        when {
            params {
                outdir = "results"
            }
        }

        then {
            assert workflow.success
        }

    }

}
