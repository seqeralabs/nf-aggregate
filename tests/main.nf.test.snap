{
    "-profile test": {
        "content": [
            10,
            {
                "PLOT_RUN_GANTT": {
                    "python": "3.9.19",
                    "pandas": "1.1.5",
                    "plotly_express": "0.4.1",
                    "click": "8.0.1"
                },
                "SEQERA_RUNS_DUMP": {
                    "seqera-cli": "0.11.2 (build 2a4e8d5)"
                },
                "Workflow": {
                    "seqeralabs/nf-aggregate": "0.6.0"
                }
            },
            [
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_general_stats.txt",
                "multiqc/multiqc_data/multiqc_seqera_cli.txt",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_data/seqera_cli_cost_plot.txt",
                "multiqc/multiqc_data/seqera_cli_cpu_time_plot.txt",
                "multiqc/multiqc_data/seqera_cli_process_statuses_plot.txt",
                "multiqc/multiqc_data/seqera_cli_wall_time_plot.txt",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_plots/pdf",
                "multiqc/multiqc_plots/pdf/general_stats_table.pdf",
                "multiqc/multiqc_plots/pdf/seqera_cli_cost_plot.pdf",
                "multiqc/multiqc_plots/pdf/seqera_cli_cpu_time_plot.pdf",
                "multiqc/multiqc_plots/pdf/seqera_cli_process_statuses_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/seqera_cli_process_statuses_plot-pct.pdf",
                "multiqc/multiqc_plots/pdf/seqera_cli_wall_time_plot.pdf",
                "multiqc/multiqc_plots/png",
                "multiqc/multiqc_plots/png/general_stats_table.png",
                "multiqc/multiqc_plots/png/seqera_cli_cost_plot.png",
                "multiqc/multiqc_plots/png/seqera_cli_cpu_time_plot.png",
                "multiqc/multiqc_plots/png/seqera_cli_process_statuses_plot-cnt.png",
                "multiqc/multiqc_plots/png/seqera_cli_process_statuses_plot-pct.png",
                "multiqc/multiqc_plots/png/seqera_cli_wall_time_plot.png",
                "multiqc/multiqc_plots/svg",
                "multiqc/multiqc_plots/svg/general_stats_table.svg",
                "multiqc/multiqc_plots/svg/seqera_cli_cost_plot.svg",
                "multiqc/multiqc_plots/svg/seqera_cli_cpu_time_plot.svg",
                "multiqc/multiqc_plots/svg/seqera_cli_process_statuses_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/seqera_cli_process_statuses_plot-pct.svg",
                "multiqc/multiqc_plots/svg/seqera_cli_wall_time_plot.svg",
                "multiqc/multiqc_report.html",
                "nf-core_ampliseq",
                "nf-core_ampliseq/gantt",
                "nf-core_ampliseq/gantt/2lXd1j7OwZVfxh_gantt.html",
                "nf-core_ampliseq/runs_dump",
                "nf-core_ampliseq/runs_dump/2lXd1j7OwZVfxh",
                "nf-core_ampliseq/runs_dump/2lXd1j7OwZVfxh/service-info.json",
                "nf-core_ampliseq/runs_dump/2lXd1j7OwZVfxh/workflow-launch.json",
                "nf-core_ampliseq/runs_dump/2lXd1j7OwZVfxh/workflow-load.json",
                "nf-core_ampliseq/runs_dump/2lXd1j7OwZVfxh/workflow-metadata.json",
                "nf-core_ampliseq/runs_dump/2lXd1j7OwZVfxh/workflow-metrics.json",
                "nf-core_ampliseq/runs_dump/2lXd1j7OwZVfxh/workflow-tasks.json",
                "nf-core_ampliseq/runs_dump/2lXd1j7OwZVfxh/workflow.json",
                "nf-core_rnaseq",
                "nf-core_rnaseq/gantt",
                "nf-core_rnaseq/gantt/4Bi5xBK6E2Nbhj_gantt.html",
                "nf-core_rnaseq/gantt/4LWT4uaXDaGcDY_gantt.html",
                "nf-core_rnaseq/runs_dump",
                "nf-core_rnaseq/runs_dump/3iFMo0NtH1Byvy",
                "nf-core_rnaseq/runs_dump/3iFMo0NtH1Byvy/service-info.json",
                "nf-core_rnaseq/runs_dump/3iFMo0NtH1Byvy/workflow-launch.json",
                "nf-core_rnaseq/runs_dump/3iFMo0NtH1Byvy/workflow-load.json",
                "nf-core_rnaseq/runs_dump/3iFMo0NtH1Byvy/workflow-metadata.json",
                "nf-core_rnaseq/runs_dump/3iFMo0NtH1Byvy/workflow-metrics.json",
                "nf-core_rnaseq/runs_dump/3iFMo0NtH1Byvy/workflow-tasks.json",
                "nf-core_rnaseq/runs_dump/3iFMo0NtH1Byvy/workflow.json",
                "nf-core_rnaseq/runs_dump/4Bi5xBK6E2Nbhj",
                "nf-core_rnaseq/runs_dump/4Bi5xBK6E2Nbhj/service-info.json",
                "nf-core_rnaseq/runs_dump/4Bi5xBK6E2Nbhj/workflow-launch.json",
                "nf-core_rnaseq/runs_dump/4Bi5xBK6E2Nbhj/workflow-load.json",
                "nf-core_rnaseq/runs_dump/4Bi5xBK6E2Nbhj/workflow-metadata.json",
                "nf-core_rnaseq/runs_dump/4Bi5xBK6E2Nbhj/workflow-metrics.json",
                "nf-core_rnaseq/runs_dump/4Bi5xBK6E2Nbhj/workflow-tasks.json",
                "nf-core_rnaseq/runs_dump/4Bi5xBK6E2Nbhj/workflow.json",
                "nf-core_rnaseq/runs_dump/4LWT4uaXDaGcDY",
                "nf-core_rnaseq/runs_dump/4LWT4uaXDaGcDY/service-info.json",
                "nf-core_rnaseq/runs_dump/4LWT4uaXDaGcDY/workflow-launch.json",
                "nf-core_rnaseq/runs_dump/4LWT4uaXDaGcDY/workflow-load.json",
                "nf-core_rnaseq/runs_dump/4LWT4uaXDaGcDY/workflow-metadata.json",
                "nf-core_rnaseq/runs_dump/4LWT4uaXDaGcDY/workflow-metrics.json",
                "nf-core_rnaseq/runs_dump/4LWT4uaXDaGcDY/workflow-tasks.json",
                "nf-core_rnaseq/runs_dump/4LWT4uaXDaGcDY/workflow.json",
                "nf-core_scrnaseq",
                "nf-core_scrnaseq/gantt",
                "nf-core_scrnaseq/gantt/38QXz4OfQDpwOV_gantt.html",
                "nf-core_scrnaseq/runs_dump",
                "nf-core_scrnaseq/runs_dump/38QXz4OfQDpwOV",
                "nf-core_scrnaseq/runs_dump/38QXz4OfQDpwOV/service-info.json",
                "nf-core_scrnaseq/runs_dump/38QXz4OfQDpwOV/workflow-launch.json",
                "nf-core_scrnaseq/runs_dump/38QXz4OfQDpwOV/workflow-load.json",
                "nf-core_scrnaseq/runs_dump/38QXz4OfQDpwOV/workflow-metadata.json",
                "nf-core_scrnaseq/runs_dump/38QXz4OfQDpwOV/workflow-metrics.json",
                "nf-core_scrnaseq/runs_dump/38QXz4OfQDpwOV/workflow-tasks.json",
                "nf-core_scrnaseq/runs_dump/38QXz4OfQDpwOV/workflow.json",
                "pipeline_info",
                "pipeline_info/collated_software_mqc_versions.yml"
            ],
            [
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
                "seqera_cli_cost_plot.txt:md5,5f6c0eec1ee86691a6d2914e163a5767",
                "seqera_cli_cpu_time_plot.txt:md5,390f4051d6092a84a2b3f73ab10643c5",
                "seqera_cli_process_statuses_plot.txt:md5,13414743798b07baebe846fb8d20bdde",
                "seqera_cli_wall_time_plot.txt:md5,abc859f5fe0dabfcded8c658de70725a",
                "workflow-launch.json:md5,e628f293acd704c57ed7ad7c3432a73e",
                "workflow-load.json:md5,947517e30d5485ad97601923f2ccb2f5",
                "workflow-metadata.json:md5,5c83e4ebc83982c35a9f8378ea984ce5",
                "workflow-metrics.json:md5,a862f20aabadbc5ab0be91642dffc00c",
                "workflow-tasks.json:md5,359044869a60162d6dea94eb1759422e",
                "workflow.json:md5,7c341a64f87ffae84b1024188184f4b2",
                "workflow-launch.json:md5,63103791dfe1fa048e8ae17a7bd27447",
                "workflow-load.json:md5,7fe53d3eb1bd3e8ab882d011a751b832",
                "workflow-metadata.json:md5,a536a5c829bd0a4441b647eef3fc10ec",
                "workflow-metrics.json:md5,15b8927b842c6fd1fc738af2e8d3462e",
                "workflow-tasks.json:md5,aea381edad7df39e2828ce63ba456c86",
                "workflow.json:md5,66bb892f5323f66b33be0dffb5a19112",
                "workflow-launch.json:md5,44c727a7586ffc5db92f8f94e0a56a90",
                "workflow-load.json:md5,4f02d5a24ab89aa648cd4346785c8f2c",
                "workflow-metadata.json:md5,207d9db4477cb7e205fb54b98206bf41",
                "workflow-metrics.json:md5,13a5b5d7447fad4a8baa053d1abf85e5",
                "workflow-tasks.json:md5,e327f5aae611c7f731a470bd7390d967",
                "workflow.json:md5,59f8a9ea713a57ea48470ff0f92cc7e5",
                "workflow-launch.json:md5,35fe5e90f51f101b22e4f0683bdf2184",
                "workflow-load.json:md5,2f5abdb8efc1fdb249378cb1af093a4a",
                "workflow-metadata.json:md5,25f9e1ffcf312b5683e1b6c1adb47726",
                "workflow-metrics.json:md5,5cb0d8e859991831eff666eed1c97b3e",
                "workflow-tasks.json:md5,e4b3c44af8c6c2663d767c8b009fa959",
                "workflow.json:md5,ade88ad661d37554229e75fc1d2b9ee4",
                "workflow-launch.json:md5,43ff8fd9b6217bb6eee542b354a7ebde",
                "workflow-load.json:md5,bebdd4917d3b769a656744ba25241898",
                "workflow-metadata.json:md5,afd35f96d1eb85bc5aedfd6f04b6f2a7",
                "workflow-metrics.json:md5,fa417298bc5c9c73c1c2bc1daf6e0406",
                "workflow-tasks.json:md5,10d3c1d5f0ff578c64702e9649e57c3a",
                "workflow.json:md5,63adb36ad806a5d6ba5ebaed7866a3d3"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-30T02:04:58.464733"
    },
    "-profile test_benchmark": {
        "content": [
            1,
            {
                "BENCHMARK_REPORT": {
                    "r": "4.4.2",
                    "quarto-cli": "1.5.55"
                },
                "Workflow": {
                    "seqeralabs/nf-aggregate": "0.6.0"
                }
            },
            [
                "benchmark_report",
                "benchmark_report/benchmark_report.html",
                "pipeline_info",
                "pipeline_info/collated_software_mqc_versions.yml"
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-30T02:09:18.441826"
    }
}